plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/TransactionScopeCachingDirectoryListerFactory.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.fs;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.cache.Cache;
+import io.airlift.units.DataSize;
+import io.trino.collect.cache.EvictableCacheBuilder;
+import io.trino.plugin.hive.HiveConfig;
+import io.trino.plugin.hive.fs.TransactionScopeCachingDirectoryLister.FetchingValueHolder;
+
+import javax.inject.Inject;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static java.lang.Math.toIntExact;
+import static java.util.Objects.requireNonNull;
+
+public class TransactionScopeCachingDirectoryListerFactory
+{
+    //TODO use a cache key based on Path & SchemaTableName and iterate over the cache keys
+    // to deal more efficiently with cache invalidation scenarios for partitioned tables.
+    private final Optional<Cache<TransactionDirectoryListingCacheKey, FetchingValueHolder>> cache;
+    private final AtomicLong nextTransactionId = new AtomicLong();
+
+    @Inject
+    public TransactionScopeCachingDirectoryListerFactory(HiveConfig hiveConfig)
+    {
+        this(requireNonNull(hiveConfig, "hiveConfig is null").getPerTransactionFileStatusCacheMaxRetainedSize(), Optional.empty());
```

Comment:
Should we rename the config and update docs if the meaning of it is now changing from being a per-transaction limit to a cumulative limit for all transactions ?

Reply:
I've updated the doc. I don't think we need to change the config name again.